pkg/credentialprovider/provider.go

```diff
@@ -83,7 +83,7 @@ func (d *defaultDockerConfigProvider) Enabled() bool {
 // Provide implements dockerConfigProvider
 func (d *defaultDockerConfigProvider) Provide() DockerConfig {
 	// Read the standard Docker credentials from .dockercfg
-	if cfg, err := ReadDockerConfigFile(); err == nil {
+	if cfg, err := ReadDockerConfigFile(nil); err == nil {
```

Comment:
are you saying don't change the signature of ReadDockerConfigFile()?  Because this invocation has to change, due to the other refactors in this PR.

If you don't want the signature changed, can you provide a reason?  I felt ok about these changes, basically allowing you to specify a set of paths to search for the dockerconfig, or pass nil if you want to search the default paths(which was the previous behavior)